### PR TITLE
fix adjusting balances after interest charged

### LINKED ISSUE #5530: adjust balances after interest accrued failing because key not found
## Describe the bug

Observed getting `Promise.reject("Error: key (a string) not found in collection \"orderedVaultStore\"")` when a vault balance was adjusted. Code shows that while the key function requires normalized debts, the values passed in are current debts.

## Expected behavior

Key always found.

### PR DESCRIPTION
**Pull request by turadg**
closes: #5530

## Description

Fixes #5530

We did have a test that charged interest and then adjusted balances: https://github.com/Agoric/agoric-sdk/blob/4c9e50ac3a9420e18b789adb618c4503af097b1c/packages/run-protocol/test/vaultFactory/test-vault-interest.js#L112-L125

But the error required exercising `updateVaultAccounting` and the test ran https://github.com/Agoric/agoric-sdk/blob/4c9e50ac3a9420e18b789adb618c4503af097b1c/packages/run-protocol/test/vaultFactory/vault-contract-wrapper.js#L143-L145
Adds to the motivation to remove that wrapper. https://github.com/Agoric/agoric-sdk/issues/5443 (h/t Hibbert)

I made a new test case in `test-vaultFactory` which reproed. Then I added a static type `NormalizedDebt` to find exactly where the wrong kind of value was being passed in.

### Security Considerations

--
### Documentation Considerations

Will land as merge commit.

### Testing Considerations

New test.

### CODE REVIEW

**Comment by Chris-Hibbert** [comment]
#5443 calls for fixing `test-vault.js`. That's the primary client of `vault-contract-wrapper.js`, and the most likely fix is to destroy `vault-contract-wrapper.js` and test against the actual contract.

File: packages/run-protocol/src/vaultFactory/vault.js [hunk 1]
```diff
@@ -240,21 +242,21 @@ const helperBehavior = {
    * maintain aggregate debt and liquidation order.
    *
    * @param {MethodContext} context
-   * @param {Amount} oldDebt - prior principal and all accrued interest
-   * @param {Amount} oldCollateral - actual collateral
-   * @param {Amount} newDebt - actual principal and all accrued interest
+   * @param {NormalizedDebt} oldDebtNormalized - prior principal and all accrued interest, normalized to time zero
```
Comment (Chris-Hibbert) [inline packages/run-protocol/src/vaultFactory/vault.js]:
```suggestion
   * @param {NormalizedDebt} oldDebtNormalized - prior principal and all accrued interest, normalized to the launch of the vaultManager
```

File: packages/run-protocol/test/vaultFactory/test-vaultFactory.js [hunk 2]
```diff
@@ -1299,6 +1299,67 @@ test('adjust balances', async t => {
   });
 });
 
+test.only('adjust balances after interest charges', async t => {
```
Comment (Chris-Hibbert) [inline packages/run-protocol/test/vaultFactory/test-vaultFactory.js]:
```suggestion
test('adjust balances after interest charges', async t => {
```

**Review by Chris-Hibbert — APPROVED** [review]
Looks good!  thanks for handling this quickly.

File: packages/run-protocol/test/vaultFactory/test-vaultFactory.js (hunk 2 above)
Comment (turadg) [reply packages/run-protocol/test/vaultFactory/test-vaultFactory.js]: ty. https://github.com/Agoric/agoric-sdk/issues/4941 is off the burner

**Pull request merged**